core/genesis/genesis.go

```diff
@@ -0,0 +1,30 @@
+package genesis
+
+import (
+	"io"
+)
+
+// Source is a source for genesis data in JSON format. It may abstract over a
+// single JSON object or separate files for each field in a JSON object that can
+// be streamed over. Modules should open a separate reader for each field that
+// is required. When fields represent arrays they can efficiently be streamed
+// over.
+type Source interface {
```

Comment:
If we're going with streaming only, maybe this is enough?

```suggestion
type Source func (field string) (io.ReadCloser, error)
```

Reply:
I don't have a strong preference but lean slightly towards interface in case we wanted to extend somehow in the future